lisa/core/customScript.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+from hashlib import md5
+from pathlib import Path
+from typing import TYPE_CHECKING, List, Optional, Type
+
+from lisa.core.tool import Tool
+from lisa.util.executableResult import ExecutableResult
+from lisa.util.process import Process
+
+if TYPE_CHECKING:
+    from lisa.core.node import Node
+
+
+class CustomScript:
+    def __init__(self, node: Node, cwd: Path, command: str) -> None:
+        self.node = node
+        self.cwd = cwd
+        self.command = command
+
+    def runAsync(
+        self,
+        extraParameters: str = "",
+        noErrorLog: bool = False,
+        noInfoLog: bool = False,
+    ) -> Process:
+        command = f"{self.command} {extraParameters}"
+        return self.node.executeAsync(
+            cmd=command,
+            useBash=True,
+            noErrorLog=noErrorLog,
+            noInfoLog=noInfoLog,
+            cwd=self.cwd,
+        )
+
+    def run(
+        self,
+        extraParameters: str = "",
+        noErrorLog: bool = False,
+        noInfoLog: bool = False,
+    ) -> ExecutableResult:
+        process = self.runAsync(
+            extraParameters=extraParameters, noErrorLog=noErrorLog, noInfoLog=noInfoLog,
+        )
+        return process.waitResult()
+
+
+class CustomScriptSpec:
+    def __init__(
+        self,
+        root_path: Path,
+        files: List[str],
+        command: Optional[str] = None,
+        dependencies: Optional[List[Type[Tool]]] = None,
```

Comment:
If instead of `None` we make this just `List[Tool] = []` (that is, defaulting to the empty list) we don't have to check `if dependencies` because we'll just iterate over an empty list.

Reply:
I tried `List[Type[Tool]]] = []`, the black gives me an error, and I couldn't get it.. So I have to change it to None. BTW, even we write like =[], it doesn't prevent None passed in. The type checking is only static checking, doesn't enforce anything on runtime.

Reply:
The message is as blow, it look ask to set it to None?

`.\lisa\core\customScript.py:54:42: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.`